src/sa_web/static/js/views/map-view.js

```diff
@@ -298,8 +316,22 @@ var Shareabouts = Shareabouts || {};
     },
 
     clearFilter: function() {
+      console.log("clearFilter");
+
       var self = this;
       this.locationTypeFilter = null;
+
+      // REFACTOR
+      // Not too sure about this part of the code... I replaced the commented-out section below
+      // with to loops that iterate through all collections in the collection object, and all models
+      // within those collections
+      _.each(this.collection, function(collection) {
+        _.each(collection.models, function(model) {
+          self.layerViews[model.cid].render();
+        });
+      });
+
+      /*
```

Comment:
Yep, that makes sense. We are no longer using this filter, because it really needs to be refactored and re-implemented. It would be helpful if you to keep an eye out for it, and let me know if it causes you any issues, or if you have any ideas for the future implementation.